libraries/Microsoft.Bot.Builder.Dialogs/WaterfallDialog.cs

```diff
@@ -20,12 +20,35 @@ public class WaterfallDialog : Dialog
         private const string StepIndex = "stepIndex";
         private const string PersistedValues = "values";
 
-        private WaterfallStep[] _steps;
+        private List<WaterfallStep> _steps;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WaterfallDialog"/> class.
+        /// </summary>
+        /// <param name="dialogId">dialog id</param>
+        /// <param name="steps">optional steps to be defined by caller</param>
         public WaterfallDialog(string dialogId, WaterfallStep[] steps = null)
```

Comment:
+1 to what Carlos is saying. I would either:
1. Have this class take a list. 
2. Use the Params to devs can just pass in "Step1, Step2, Step3". 

I seldom see C# Developers using arrays...